worlds/Files.py

```diff
@@ -137,7 +137,17 @@ def read(self, file: Optional[Union[str, BinaryIO]] = None) -> None:
                 raise InvalidDataError(f"{message}This might be the incorrect world version for this file") from e
 
     def read_contents(self, opened_zipfile: zipfile.ZipFile) -> Dict[str, Any]:
-        with opened_zipfile.open("archipelago.json", "r") as f:
+        try:
+            manifest_info = opened_zipfile.getinfo(self.manifest_path)
```

Comment:
```suggestion
        try:
            manifest_info = opened_zipfile.getinfo("archipelago.json")
```

Reply:
```suggestion
        if not self.manifest_path.endswith("archipelago.json"):
            raise ValueError("Manifest path should end with archipelago.json")
        try:
            manifest_info = opened_zipfile.getinfo(self.manifest_path)
```

Reply:
Probably good enough 

Someone will do sth like "thisfileendswitharchipelago.json", but at that point they're breaking stuff on purpose, and are one step away from just overriding get_manifest